### PR TITLE
Model XLA Flags

### PR DESCRIPTION
**Pull request by terrykong**
Moves XLA flags from model CI into their own files that can be sourced. Each file can be sourced and will print what it sets. 

Some files source other files, which was intentional to avoid introducing sim-links into the repo, which can sometimes have platform issues (like on windows).

### CODE REVIEW

File: rosetta/rosetta/projects/pax/xla_flags/grok-proxy.env [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    --xla_gpu_all_gather_combine_threshold_bytes=3221225472 \
+    --xla_gpu_reduce_scatter_combine_threshold_bytes=402653184 \
+    --xla_gpu_enable_pipelined_all_gather=true \
+    --xla_gpu_enable_pipelined_reduce_scatter=true \
+    --xla_gpu_enable_pipelined_all_reduce=true \
+    --xla_gpu_enable_while_loop_double_buffering=true \
+    --xla_gpu_enable_all_gather_combine_by_dim=false \
+    --xla_gpu_enable_reduce_scatter_combine_by_dim=false \
+    --xla_disable_hlo_passes=rematerialization \
+    --xla_gpu_enable_custom_fusions=true
```
[the remark this answers is not in the document]
Comment (terrykong) [reply rosetta/rosetta/projects/pax/xla_flags/grok-proxy.env:18]: These were pulled from model CI tests. By moving them into JAX-Toolbox I'm hoping they can be updated in sync with documentation changes and we can then trigger the internal CI with PRs that change these xla flag files.

I haven't seen any failures with these flags yet with the latest nightlies. Is it okay to tackle that in a followup given these aren't linked to any docs yet?

File: rosetta/rosetta/projects/pax/xla_flags/llama-7b.env [hunk 2]
```diff
@@ -0,0 +1,4 @@
+set -x
+echo "$0 uses default XLA_FLAGS='${XLA_FLAGS:-}'"
```
[the remark this answers is not in the document]
Comment (terrykong) [reply rosetta/rosetta/projects/pax/xla_flags/llama-7b.env:2]: done

File: rosetta/rosetta/projects/pax/xla_flags/glam-126m64e.env [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# sources 5b as a reference
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+source $SCRIPT_DIR/gpt-5b.env
+unset $SCRIPT_DIR
```
Comment (DwarKapex) [inline rosetta/rosetta/projects/pax/xla_flags/glam-126m64e.env:4]: Remove `$`?

File: rosetta/rosetta/projects/pax/xla_flags/glam-64b64e.env [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# sources 5b as a reference
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+source $SCRIPT_DIR/gpt-5b.env
+unset $SCRIPT_DIR
```
Comment (DwarKapex) [inline rosetta/rosetta/projects/pax/xla_flags/glam-64b64e.env:4]: Remove `$`?

File: rosetta/rosetta/projects/pax/xla_flags/gpt-175b.env [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# sources 5b as a reference
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+source $SCRIPT_DIR/gpt-5b.env
+unset $SCRIPT_DIR
```
Comment (DwarKapex) [inline rosetta/rosetta/projects/pax/xla_flags/gpt-175b.env:4]: Remove `$`?

File: rosetta/rosetta/projects/pax/xla_flags/gpt-126m.env [hunk 6]
```diff
@@ -0,0 +1,12 @@
+set -x
+export XLA_FLAGS="\
+    --xla_gpu_enable_latency_hiding_scheduler=true \
+    --xla_allow_excess_precision \
+    --xla_gpu_enable_highest_priority_async_stream=true \
+    --xla_gpu_enable_triton_softmax_fusion=false \
+    --xla_gpu_all_reduce_combine_threshold_bytes=33554432 \
```
Comment (DwarKapex) [inline rosetta/rosetta/projects/pax/xla_flags/gpt-126m.env:7]: Maybe add `THRESHOLD_BYTES` as above?

File: rosetta/rosetta/projects/pax/xla_flags/gpt-126m.env [hunk 7]
```diff
@@ -0,0 +1,12 @@
+set -x
+export XLA_FLAGS="\
+    --xla_gpu_enable_latency_hiding_scheduler=true \
+    --xla_allow_excess_precision \
```
Comment (DwarKapex) [inline rosetta/rosetta/projects/pax/xla_flags/gpt-126m.env:4]: Its color is different with respect to other values. Could you please check whitespaces around?

File: rosetta/rosetta/projects/pax/xla_flags/gpt-5b.env [hunk 8]
```diff
@@ -0,0 +1,11 @@
+set -x
+export XLA_FLAGS="\
+    --xla_gpu_enable_latency_hiding_scheduler=true \
+    --xla_allow_excess_precision \
```
Comment (DwarKapex) [inline rosetta/rosetta/projects/pax/xla_flags/gpt-5b.env:4]: Same here as above

File: rosetta/rosetta/projects/pax/xla_flags/llama-70b.env [hunk 9]
```diff
@@ -0,0 +1,4 @@
+# sources 5b as a reference
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+source $SCRIPT_DIR/gpt-5b.env
+unset $SCRIPT_DIR
```
Comment (DwarKapex) [inline rosetta/rosetta/projects/pax/xla_flags/llama-70b.env:4]: Remove `$`?

File: rosetta/rosetta/projects/pax/xla_flags/glam-126m64e.env (hunk 3 above)
Comment (terrykong) [reply rosetta/rosetta/projects/pax/xla_flags/glam-126m64e.env]: thanks, good catch. done

File: rosetta/rosetta/projects/pax/xla_flags/gpt-126m.env (hunk 6 above)
Comment (terrykong) [reply rosetta/rosetta/projects/pax/xla_flags/gpt-126m.env]: done

File: rosetta/rosetta/projects/pax/xla_flags/gpt-5b.env
Comment (ashors1) [inline rosetta/rosetta/projects/pax/xla_flags/gpt-5b.env:1]: Would it make sense to rename this to something like `common.env` to reflect the fact that these flags are shared across many models?

Comment (terrykong) [reply rosetta/rosetta/projects/pax/xla_flags/gpt-5b.env:1]: Yup, great suggestion. Done!

File: rosetta/rosetta/projects/pax/xla_flags/grok-proxy.env (hunk 1 above)
[the remark this answers is not in the document]
Comment (nouiz) [reply rosetta/rosetta/projects/pax/xla_flags/grok-proxy.env:21]: SG.

**Review by ashors1 — APPROVED** [review]

**Pull request merged**